src/notation/qml/MuseScore/NotationScene/internal/TextStylePopup.qml

```diff
@@ -0,0 +1,389 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2023 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import MuseScore.Ui 1.0
+import MuseScore.UiComponents 1.0
+import MuseScore.NotationScene 1.0
+import MuseScore.Inspector 1.0
+
+StyledPopupView {
+    id: root
+
+    property NavigationSection notationViewNavigationSection: null
+    property QtObject model: null
+    property int navigationOrderStart: 0
+    property int navigationOrderEnd: 0
+
+    contentWidth: menuItems.width
+
+    contentHeight: menuItems.height
+
+    margins: 10
+
+    showArrow: false
+
+    function updatePosition(pos, size) {
+        var h = 150
+        root.x = pos.x + size.x + 12
+        root.y = pos.y - h / 2
+    }
+
+
+    TextStylePopupModel {
+        id: textStyleModel
+
+        onItemRectChanged: function(rect) {
+            updatePosition(Qt.point(rect.x, rect.y), Qt.point(rect.width, rect.height))
+        }
+    }
+
+    Component.onCompleted: {
+        textStyleModel.init()
+    }
+
+    //The main popup
```

Comment:
I'd be a bit more sparing with comments like this, and watch out that they don't clutter the file. This one for example, is not really necessary.

You can also make them a bit smaller: for example, instead of 
```
            //Column -> 1 | Row -> 1
            RowLayout {
```
you can write 
```
            RowLayout { // column 1, row 1
```

And comments for individual controls are often not necessary either because their `id` already says enough.